pkg/testcase/ingressdns.go

```diff
@@ -59,3 +63,68 @@ func TestDnsAccess(deleteWorkload bool) {
 		Expect(err).NotTo(HaveOccurred(), "dnsutils manifest not deleted")
 	}
 }
+
+func TestIngressRoute(deleteWorkload bool) {
+	// Update base IngressRoute manifest to use one of the Node External IPs
+	originalFilePath := shared.BasePath() +
+		fmt.Sprintf("/distros-test-framework/workloads/%s/ingressroute.yaml", shared.Arch)
+	newFilePath := shared.BasePath() + 
+		fmt.Sprintf("/distros-test-framework/workloads/%s/dynamic-ingressroute.yaml", shared.Arch)
+	content, err := os.ReadFile(originalFilePath)
+	if err != nil {
+		Expect(err).NotTo(HaveOccurred(), "failed to read file for ingressroute resource")
+	}
+
+	publicIp := fmt.Sprintf("%s.nip.io", shared.FetchNodeExternalIP()[0])
+	newContent := strings.ReplaceAll(string(content), "$YOURDNS", publicIp)
+	err = os.WriteFile(newFilePath, []byte(newContent), 0644)
+	if err != nil {
+		Expect(err).NotTo(HaveOccurred(), "failed to update file for ingressroute resource to use one of the node external ips")
+	}
+
+	// Deploy manifest and ensure pods are running
+	_, err = shared.ManageWorkload("apply", "dynamic-ingressroute.yaml")
+	Expect(err).NotTo(HaveOccurred(), "IngressRoute manifest not successfully deployed")
+
+	getIngressRoutePodsRunning := fmt.Sprintf("kubectl get pods -n test-ingressroute -l app=whoami" +
+		" --field-selector=status.phase=Running --kubeconfig=%s", shared.KubeConfigFile)
+	err = assert.ValidateOnHost(getIngressRoutePodsRunning, statusRunning)
+	Expect(err).NotTo(HaveOccurred(), err)
+
+	// Query the IngressRoute Host
+	pods, err := shared.GetPodsByNamespaceAndLabel("test-ingressroute", "app=whoami", false)
+	Expect(err).NotTo(HaveOccurred(), err)
+	
+	for _, pod := range pods {
```

Comment:
i think here we have a opportunity to improve the code + the framework

what u think about creating a func on assert pkg that uses your Getpods.... and receives one args variadic something like that 
```
func  (args ...string) 
```
or receive 2 lists for simplicite , one list for cmds and another for asserts, 

```
func  ([]cmds,[]asserts) error {

here u call RunCommandHost(cmd) that accepts list of params and validate your stuff
}
```

then here on the test case it self you will abstract a lot just calling your new function that will do all of that 
```
pods, err := shared.GetPodsByNamespaceAndLabel("test-ingressroute", "app=whoami", false)
	Expect(err).NotTo(HaveOccurred(), err)

	for _, pod := range pods {
Comment
		Eventually(func(g Gomega) {
			// Positive test cases
			err = assert.CheckComponentCmdHost("curl -sk http://"+publicIp+"/notls",
				fmt.Sprintf("Hostname: %s", pod.Name),
				fmt.Sprintf("IP: %s", pod.NodeIP),
			)
			Expect(err).NotTo(HaveOccurred(), err)

			err = assert.CheckComponentCmdHost("curl -sk https://"+publicIp+"/tls",
				fmt.Sprintf("Hostname: %s", pod.Name),
				fmt.Sprintf("IP: %s", pod.NodeIP),
			)
			Expect(err).NotTo(HaveOccurred(), err)

			// Negative test cases
			err = assert.CheckComponentCmdHost("curl -sk http://"+publicIp+"/tls",
				"404 page not found",
			)
			Expect(err).NotTo(HaveOccurred(), err)

			err = assert.CheckComponentCmdHost("curl -sk https://"+publicIp+"/notls",
				"404 page not found",
			)
			Expect(err).NotTo(HaveOccurred(), err) 
```

+ no repetion calling CheckComponentCmdHost a lot of times sequentially 



}

Reply:
I agree this makes sense. Check out the latest commit for the implementation and let me know if that satisfies what you're thinking.

Reply:
i liked but since we do already have 2 functions that take care of that as  assert.CheckComponentCmdHost and assert.ValidateOnHost it will unecessary creating new one on assert. 
i misinterpreted your need probably sorry.

also it will be too duplication of responsabilities have this CheckMultipleCmdHost, just calling the CheckComponentCmdHost both already range over asserts

i will leave u with a more lean and idiomatic approach.
 

```
	// Query the IngressRoute Host
	pods, err := shared.GetPodsByNamespaceAndLabel("test-ingressroute", "app=whoami", false)
	Expect(err).NotTo(HaveOccurred(), err)

	for _, pod := range pods {
		helper := struct {
			cmds []string
			asserts
		}{
			cmds: []string{
				"curl -sk http://" + publicIp + "/notls",
				"curl -sk https://" + publicIp + "/tls",
			},
			asserts: asserts{
				positive: []string{
					fmt.Sprintf("Hostname: %s", pod.Name),
					fmt.Sprintf("IP: %s", pod.NodeIP),
				},
				negative: "404 page not found",
			},
		}

		for _, cmd := range helper.cmds {
			err = assert.CheckComponentCmdHost(cmd, helper.asserts.positive...)
			Expect(err).NotTo(HaveOccurred(), err)
			err = assert.CheckComponentCmdHost(cmd, helper.asserts.negative)
			Expect(err).To(HaveOccurred(), err)
		}
	}


```

Reply:
hmmm I see what you wanted now and the solution you provided is really nice! But in that case I think we should probably revert it to _almost how_ I had it originally as there isn't a direct need for including a new type on this. I will modify to use a string slice for the asserts though.

Reply:
See latest commit and let me know what you think -- there are actually four separate commands here (even though they're very similar). This actually has the least amount of nested loops as well so is probably the most ideal!

Reply:
for me its totally fine though, however i really prefer 2  for loops  rather rather than call 4 for times ` assert.CheckComponentCmdHost(` a function , plus imagine that you have a 15 positive and 10 negative asserts to do, it would go crazy this , thats why the usage of the struct here, to create a table driven strategy.
even though having 2 cmds only .
but again if this suits your actual need ! go for it

Reply:
Yeah let's go with this right now. We can refactor in the future if need arises!